searx/engines/cargo.py

```diff
@@ -0,0 +1,69 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""hex.pm"""
+
+from urllib.parse import urlencode
+from dateutil import parser
+from flask_babel import gettext
+
+
+about = {
+    # pylint: disable=line-too-long
+    "website": "https://crates.io/",
+    "wikidata_id": None,
+    "official_api_documentation": "https://crates.io/data-access",
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": "JSON",
+}
+
+categories = ["it", "packages"]
+
+
+# engine dependent config
+paging = True
+search_url = "https://crates.io/api/v1/crates"
+
+
+def request(query: str, params):
+    args = urlencode({"page": params["pageno"], "q": query, "per_page": 20})
+    params["url"] = f"{search_url}?{args}"
+    return params
+
+
+def response(resp):
+    results = []
+    for package in resp.json()["crates"]:
+        published_date = package.get("updated_at")
+        published_date = parser.parse(published_date)
+        links = get_links(package)
+
+        results.append(
+            {
+                "template": "packages.html",
+                "url": f'https://crates.io/crates/{package["name"]}',
+                "title": package["name"],
+                "package_name": package["name"],
+                "tags": package["keywords"],
+                "content": package["description"],
+                "version": package["max_version"],
+                "publishedDate": published_date,
+                "links": links,
+            }
+        )
+
+    return results
+
+
+def get_links(package):
+    docs_key = gettext("Documentation")
+    homepage_key = gettext("Project homepage")
+    source_code_key = gettext("Source code")
+    links = {}
+    if package.get("documentation"):
+        links[docs_key] = package["documentation"]
+    if package.get("homepage"):
+        links[homepage_key] = package["homepage"]
+    if package.get("repository"):
+        links[source_code_key] = package["repository"]
+    if links:
+        return links
```

Comment:
This should probably return `links`, no matter if it's empty or not (not sure how Python handles functions without return values)